server/plugin/api_test.go

```diff
@@ -1110,135 +1071,165 @@ func TestHandlePipelineApproveOrRejectReleaseRequest(t *testing.T) {
+
+			w := httptest.NewRecorder()
+			p.handlePipelineApproveOrRejectReleaseRequest(w, req)
+			resp := w.Result()
+			assert.Equal(t, testCase.statusCode, resp.StatusCode)
+		})
+	}
+}
+
+func TestHandlePipelineCommentModal(t *testing.T) {
+	defer monkey.UnpatchAll()
+	mockAPI := &plugintest.API{}
+	mockCtrl := gomock.NewController(t)
+	mockedClient := mocks.NewMockClient(mockCtrl)
+	p := setupMockPlugin(mockAPI, nil, mockedClient)
+	for _, testCase := range []struct {
+		description      string
+		body             string
+		statusCode       int
+		isPayloadInvalid bool
+		openModalError   error
+	}{
+		{
+			description: "HandlePipelineCommentModal: valid release approval confirmation",
 			body: `{
 				"post_id": "mockPostID",
 				"channel_id": "mockChannelID",
 				"context": {
 				  "approvalId": 1234,
+				  "requestName": "release",
 				  "organization": "mockOrganization",
 				  "projectName": "mockProjectName",
 				  "requestType": "mockRequestType"
 				}
 			  }`,
-			updatePipelineApprovalRequestStatus: http.StatusBadRequest,
-			getApprovalDetailsError:             errors.New("failed to get approval details"),
-			statusCode:                          http.StatusInternalServerError,
-			getApprovalDetailsStatus:            http.StatusInternalServerError,
+			statusCode: http.StatusOK,
 		},
 		{
-			description: "HandlePipelineApproveOrRejectReleaseRequest: invalid payload",
+			description: "HandlePipelineCommentModal: valid run approval confirmation",
 			body: `{
 				"post_id": "mockPostID",
 				"channel_id": "mockChannelID",
 				"context": {
-				  "approvalId": 1234,
-				  "wrong: "mockOrganization",
-				  "projectName": "mockProjectName",
+				  "approvalId": "mockApprovalID",
+				  "requestName": "run",
+				  "organization": "mockOrganization",
+				  "projectId": "mockProjectID",
 				  "requestType": "mockRequestType"
 				}
 			  }`,
-			isPayloadInvalid: true,
-			statusCode:       http.StatusInternalServerError,
+			statusCode: http.StatusOK,
 		},
 		{
-			description: "HandlePipelineApproveOrRejectReleaseRequest: failed to approve/reject the request due to server error",
+			description: "HandlePipelineCommentModal: failed to open comment modal",
 			body: `{
 				"post_id": "mockPostID",
 				"channel_id": "mockChannelID",
 				"context": {
-				  "approvalId": 1234,
+				  "approvalId": "mockApprovalID",
+				  "requestName": "run",
 				  "organization": "mockOrganization",
-				  "projectName": "mockProjectName",
+				  "projectId": "mockProjectID",
 				  "requestType": "mockRequestType"
 				}
 			  }`,
-			updatePipelineApprovalRequestStatus:       http.StatusInternalServerError,
-			statusCode:                                http.StatusInternalServerError,
-			updatePipelineReleaseApprovalRequestError: errors.New("failed to update the pipeline approval request"),
+			statusCode:     http.StatusInternalServerError,
+			openModalError: errors.New("failed to open comment modal"),
```

Comment:
check this everywhere
```suggestion
			openModalError: errors.New("failed to open the comment modal"),
```